invenio_administration/assets/semantic-ui/js/invenio_administration/details/AdminDetailsView.js

```diff
@@ -0,0 +1,131 @@
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import { Grid, Header, Segment, Dimmer, Loader } from "semantic-ui-react";
+import DetailsComponent from "./DetailsComponent";
+import DetailsActions from "./DetailsActions";
+import { InvenioAdministrationActionsApi } from "../api/actions";
+import AdminDetailsContext from "./AdminDetailsContext";
+
+export default class AdminDetailsView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      data: undefined,
+      error: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({ loading: true });
+    const { apiEndpoint, pid } = this.props;
+
+    try {
+      const response = await InvenioAdministrationActionsApi.getResource(
+        apiEndpoint,
+        pid
+      );
+
+      this.setState({
+        loading: false,
+        data: response.data,
+        error: undefined,
+      });
+    } catch (e) {
+      console.error(e);
+      //TODO
+    }
+  };
+
+  /**
+   * Sorts given columns given their provided order.
+   * Returns an array of objects, each object being composed as:
+   * {
+   *   key: str,
+   *   text: str,
+   *   order: number
+   * }
+   *
+   * @param {Object[]} columns
+   * @returns {Object[]} an array of objects, ordered by the provided order.
+   */
+  sortColumns = (columns) => {
```

Comment:
`columns` should be passed as an object like:

``` javascript
{
  id: {
    order: 1,
    text: "ID"
  }
}
```

This function converts such object into an array of objects, guaranteeing thus the order defined by the view. E.g:

```javascript
[
  {
    key: "id",
    text: "ID",
    order: 1
  }, 
  {
    key: "name",
    text: "Awesome name",
    order: 2
  }, 
]
```